browser/brave_wallet/brave_wallet_tab_helper.cc

```diff
@@ -34,6 +35,40 @@ void BraveWalletTabHelper::ShowBubble() {
   wallet_bubble_manager_delegate_->ShowBubble();
 }
 
+void BraveWalletTabHelper::UserRequestCompleted(const std::string& requestData,
+                                                const std::string& result) {
+  size_t hash = base::FastHash(base::as_bytes(base::make_span(requestData)));
+  DCHECK(request_callbacks_.count(hash));
+  std::move(request_callbacks_[hash]).Run(result);
+  request_callbacks_.erase(hash);
+}
+
+void BraveWalletTabHelper::RequestUserApproval(
+    const std::string& requestData,
+    RequestEthereumChainCallback callback) {
+  std::string requesting_origin;
+  std::vector<std::string> accounts;
+  auto* manager =
+      permissions::PermissionRequestManager::FromWebContents(web_contents_);
+  DCHECK(manager);
+
+  int32_t tab_id = sessions::SessionTabHelper::IdForTab(web_contents_).id();
+  GURL webui_url = brave_wallet::GetAddEthereumChainPayloadWebUIURL(
```

Comment:
This shouldn't be put in ethereum_permission_utils.h, it's only used in this file, you could use a anonymous function here.